packages/components/src/DataList/components/DataListLayoutInternal/DataListHeader.tsx

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import { DataListLayoutInternal } from "./DataListLayoutInternal";
+import { Breakpoints } from "../../DataList.const";
+import styles from "../../DataList.css";
+import {
+  DataListHeader as DataListHeaderType,
+  DataListItemTypeFromHeader,
+  DataListObject,
+  DataListProps,
+} from "../../DataList.types";
+import { sortSizeProp } from "../../DataList.utils";
+import { DataListLayoutProps } from "../DataListLayout/DataListLayout";
+
+export function DataListHeader<T extends DataListObject>({
+  layouts,
+  mediaMatches,
+  headerData,
+  headerVisibility,
+}: {
+  layouts: React.ReactElement<DataListLayoutProps<T>>[] | undefined;
+  mediaMatches?: Record<Breakpoints, boolean>;
+  headerData?: DataListItemTypeFromHeader<T, DataListHeaderType<T>>;
+  headerVisibility: NonNullable<DataListProps<T>["headerVisibility"]>;
+}) {
+  return (
+    <DataListLayoutInternal
+      layouts={layouts}
+      renderLayout={(layout: React.ReactElement<DataListLayoutProps<T>>) => {
+        const matchingMediaQueries = Object.keys(mediaMatches || {}).filter(
+          (key): key is Breakpoints => !!mediaMatches?.[key as Breakpoints],
+        );
+        const sortedVisibleBreakpoints = sortSizeProp(matchingMediaQueries);
+        // Determines if the header should be visible based on the headerVisibility
+        const showHeader = sortedVisibleBreakpoints.reduce(
+          (previousVisibility, breakpoint) => {
+            return headerVisibility[breakpoint] ?? previousVisibility;
+          },
+          true,
+        );
```

Comment:
Does this need to be inside this function? It seems to not rely anything on the param from the render layout